### PR TITLE
Improving makefile to support other branches, and tagging on Beta channel

### PR DESCRIPTION
**Pull request by austinchambers**
* Improved username in release notes (gets author in both github and local env)
* If on xyz branch, we release on Xyz channel with a hash and an optional "dirty" flag
* The exception to the above: When on "master" branch, we release to "Stable" channel.
* If a tag comes in, we release on Beta channel with a version matching the tag (We want all these situations to be through CI, so this isn't supported locally).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
 SHELL := /bin/bash -o pipefail
 
+app_slug := "${REPLICATED_APP}"
+release_notes := "CLI release by ${shell git log -1 --pretty=format:'%ae'} on $(shell date)"
+
+# If tag is set and we're using github_actions, that takes precedence and we release on the beta channel. 
+# Otherwise, get the branch, ensure the first char is uppercase, and use to build version. 
+ifeq ($(origin GITHUB_TAG_NAME),undefined)
+ifeq ($(origin ${GITHUB_BRANCH_NAME}),undefined)
+channel := $(shell ch=$$(git rev-parse --abbrev-ref HEAD);echo $$(tr '[:lower:]' '[:upper:]' <<< $${ch:0:1})$${ch:1})
```
Comment (dexhorthy) [inline Makefile]: I want to push back again on the Upper-casing of the first character. I think the majority of folks will use code-like tokens for branch names, like `fixes/update-spline-reticulation` or `change-some-other-thing`, and those would translate to branch names of `Change-some-other-thing` or `Fixes/update-spline-reticulation`. 

Do you think we can reliably Human-readable-ify every possible branch name? I think its much easier (and simpler) to let develops customize Casing on branches.

File: Makefile [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+channel := "Beta"
+version := ${GITHUB_TAG_NAME}
+endif
+
 .PHONY: deps-vendor-cli
-deps-vendor-cli: dist = $(shell echo `uname` | tr '[:upper:]' '[:lower:]')
-deps-vendor-cli: cli_version = ""
-deps-vendor-cli: cli_version = $(shell [[ -x deps/replicated ]] && deps/replicated version | grep version | head -n1 | cut -d: -f2 | tr -d , )
+	$(eval dist := $(shell echo `uname` | tr '[:upper:]' '[:lower:]'))
+	$(eval cli_version := $(shell [[ -x deps/replicated ]] && deps/replicated version | grep version | head -n1 | cut -d: -f2 | tr -d , ))
```
Comment (dexhorthy) [inline Makefile]: was there a functional reason for this change or is this a style thing?

File: Makefile [hunk 3]
```diff
@@ -47,7 +61,7 @@ list-releases: check-api-token check-app deps-vendor-cli
 	deps/replicated release ls --app $(app_slug)
 
 .PHONY: release
-release: check-api-token check-app deps-vendor-cli lint
+release: check-api-token check-app deps-vendor-cli
```
Comment (dexhorthy) [inline Makefile]: was removal of `lint` intentional?

File: Makefile (hunk 1 above)
Comment (austinchambers) [reply Makefile]: Oh, right, I only need to do this for Unstable. Will change. 

The real fix here is to make our damn CLI not case-sensitive. A general rule for development: very, very few things should be made case sensitive.

File: Makefile (hunk 2 above)
Comment (austinchambers) [reply Makefile]: Just exploring options. I undid this.

File: Makefile (hunk 3 above)
Comment (austinchambers) [reply Makefile]: Nice catch

**Pull request merged**